File: 2022/phase2_dress_rehearsal/byuvrx/src/frostlab-inekf-54148f8ac0c7/tests/SE2Models/test_Odometry.cpp
#include "gtest/gtest.h"
#include <Eigen/Core>
#include <InEKF/Core>
#include <InEKF/SE2Models>

#define EXPECT_MATRICES_EQ(M_actual, M_expected) \
  EXPECT_TRUE(M_actual.isApprox(M_expected, 1e-6)) << "  Actual:\n" << M_actual << "\nExpected:\n" << M_expected


TEST(OdometryProcess, f){
    InEKF::SE2<> state;
    InEKF::SE2<> U(.1, 1, 2);

    InEKF::OdometryProcess op;

    EXPECT_MATRICES_EQ(op.f(U, 1, state)(), U());
}

TEST(OdometryProcess, MakePhi){
    InEKF::SE2<> state;
    InEKF::SE2<> U(.1, 1, 2);

    InEKF::OdometryProcess op;
    Eigen::Matrix3d Phi;

    // check right
    Phi = op.MakePhi(U, 1, state, InEKF::RIGHT);
    EXPECT_MATRICES_EQ(Phi, Eigen::Matrix3d::Identity());

    // check left
    Phi = op.MakePhi(U, 1, state, InEKF::LEFT);
    EXPECT_MATRICES_EQ(Phi, U.inverse().Ad());
}

File: 2022/phase2_dress_rehearsal/byuvrx/src/frostlab-inekf-54148f8ac0c7/python/tests/SE2_SLAM/test_Odometry.py
import numpy as np
from numpy.testing import assert_allclose
from inekf import SE2, OdometryProcess, ERROR
import pytest

def test_f():
    state = SE2()
    u = SE2(.1, 1, 2)

    op = OdometryProcess()

    assert_allclose(op.f(u, 1, state).State, u.State)

def test_MakePhi():
    state = SE2()
    u = SE2(.1, 1, 2)

    op = OdometryProcess()

    phi = op.MakePhi(u, 1, state, ERROR.RIGHT)
    assert_allclose(phi, np.eye(3))

    phi = op.MakePhi(u, 1, state, ERROR.LEFT)
    assert_allclose(phi, u.inverse().Ad())